src/Adapter/Product/Update/ProductSupplierUpdater.php

```diff
@@ -111,45 +132,40 @@ public function setCombinationSuppliers(
     }
 
     /**
-     * Removes associated product suppliers
-     * If combinationId is provided, then it only removes product suppliers associated to that combination
-     * If combinationId is null, then it only removes product suppliers that are not associated to any combination
+     * Removes all product suppliers associated to specified product without combinations
      *
      * @param ProductId $productId
-     * @param CombinationId|null $combinationId
      */
-    public function removeAll(ProductId $productId, ?CombinationId $combinationId = null): void
+    public function removeAllForProduct(ProductId $productId): void
     {
         $product = $this->productRepository->get($productId);
-        $productSuppliersInfo = $this->productSupplierRepository->getProductSuppliersInfo($productId, $combinationId);
 
-        $productSupplierIds = [];
-        foreach ($productSuppliersInfo as $productSupplier) {
-            $productSupplierIds[] = new ProductSupplierId((int) $productSupplier['id_product_supplier']);
+        if ($product->cache_default_attribute) {
+            throw new RuntimeException(sprintf(
```

Comment:
I think it deserves a dedicated exception something like `InvalidProductType` or similar (same in `setProductSuppliers`)